packages/salesforcedx-vscode-apex/test/vscode-integration/commands/forceApexExecute.test.ts

```diff
@@ -0,0 +1,257 @@
+          getText: () => text,
+          isUntitled: true
+        },
+        selection: { isEmpty: true, text: 'System.assert(false);' }
+      };
+      sb.stub(vscode.window, 'activeTextEditor').get(() => {
+        return mockActiveTextEditor;
+      });
+
+      const fileNameGatherer = new AnonApexGatherer();
+      const result = (await fileNameGatherer.gather()) as ContinueResponse<{
+        apexCode: string;
+      }>;
+      expect(result.data.apexCode).to.equal(text);
+    });
+
+    it(`should return the currently highlighted 'selection' to execute anonymous apex`, async () => {
+      const mockActiveTextEditor = {
+        document: {
+          getText: (doc: { isEmpty: boolean; text: string }) => doc.text,
+          isUntitled: true
+        },
+        selection: { isEmpty: false, text: 'System.assert(true);' }
+      };
+      sb.stub(vscode.window, 'activeTextEditor').get(() => {
+        return mockActiveTextEditor;
+      });
+
+      const apexCodeGatherer = new AnonApexGatherer();
+      const result = (await apexCodeGatherer.gather()) as ContinueResponse<{
+        apexCode: string;
+      }>;
+      expect(result.data.apexCode).to.equal('System.assert(true);');
+    });
+  });
+
+  describe('use CLI Command setting', async () => {
+    let settingStub: SinonStub;
+    let apexExecutorStub: SinonStub;
+    let cliExecutorStub: SinonStub;
+    let anonGather: SinonStub;
+    let apexTempFile: SinonStub;
+
+    beforeEach(() => {
+      settingStub = sb
+        .stub()
+        .withArgs('experimental.useApexLibrary')
+        .returns(true);
+      sb.stub(vscode.workspace, 'getConfiguration').returns({
+        get: settingStub
+      });
+      apexExecutorStub = sb.stub(
+        ApexLibraryExecuteExecutor.prototype,
+        'execute'
+      );
+      cliExecutorStub = sb.stub(ForceApexExecuteExecutor.prototype, 'execute');
+      anonGather = sb
+        .stub(AnonApexGatherer.prototype, 'gather')
+        .returns({ type: 'CONTINUE' } as ContinueResponse<{}>);
+      apexTempFile = sb
+        .stub(CreateApexTempFile.prototype, 'gather')
+        .returns({ type: 'CONTINUE' } as ContinueResponse<{}>);
+    });
+
+    it('should use the ApexLibraryExecuteExecutor if setting is false', async () => {
+      settingStub.returns(true);
+      await forceApexExecute();
+      expect(apexExecutorStub.calledOnce).to.be.true;
+      expect(anonGather.calledOnce).to.be.true;
+      expect(cliExecutorStub.called).to.be.false;
+      expect(apexTempFile.called).to.be.false;
+    });
+
+    it('should use the ForceApexExecuteExecutor if setting is true', async () => {
+      settingStub.returns(false);
+      await forceApexExecute();
+      expect(cliExecutorStub.calledOnce).to.be.true;
+      expect(apexTempFile.calledOnce).to.be.true;
+      expect(apexExecutorStub.called).to.be.false;
+      expect(anonGather.called).to.be.false;
+    });
+  });
+
+  describe('Format Execute Anonymous Response', () => {
```

Comment:
Brought these tests into the command's test file since formatting results are part of the command.